api/server/server.go

```diff
@@ -790,30 +790,23 @@ func postContainersCreate(eng *engine.Engine, version version.Version, w http.Re
 		return err
 	}
 	var (
-		job          = eng.Job("create", r.Form.Get("name"))
-		outWarnings  []string
-		stdoutBuffer = bytes.NewBuffer(nil)
-		warnings     = bytes.NewBuffer(nil)
+		warnings []string
+		name     = r.Form.Get("name")
+		env      = new(engine.Env)
 	)
 
-	if err := job.DecodeEnv(r.Body); err != nil {
+	if err := env.Decode(r.Body); err != nil {
```

Comment:
@LK4D4 I didn't replace anything, job calls env internally, I removed the layer on top of the env. It's not worse, we're removing the references to job, that's good.

Replacing the job with typed structs requires to go through all the project's code base to remove the reference to env before merging this. That's the second script.